packages/rehype-dom-parse/lib/index.js

```diff
@@ -34,13 +34,13 @@ export default function parse(options) {
  */
 function createFragment(htmlString) {
   const fragment = document.createDocumentFragment()
-  const temporary = document.createElement('body')
-  temporary.innerHTML = htmlString
-  let child = temporary.firstChild
+  const doc = createDocument(`<body>${htmlString}</body>`)
 
-  while (child) {
+  /** @type {Node | null} */
+  let child
+
+  while ((child = doc.body.firstChild)) {
     fragment.append(child)
```

Comment:
```suggestion
    fragment.append(child)
    child = doc.body.firstChild
```

Reply:
Fine, although I prefer the previous one. 🤣